nautobot/core/apps/__init__.py

```diff
@@ -76,6 +78,16 @@ def register_menu_items(tab_list):
 
                 group_perms = set()
                 for item in group.items:
+
+                    # Instead of passing the reverse url strings, we pass in the url itself initialized with args and kwargs.
+                    try:
+                        item.link = reverse(item.link, args=item.args, kwargs=item.kwargs)
+                    except NoReverseMatch as e:
+                        logger = logging.getLogger("nautobot.core.apps")
+                        logger.debug(
+                            f"{e}, Please specify `args` or `kwargs` attribute in NavMenuItem in your `navigation.py` file"
+                        )
```

Comment:
Instead could we just default to empty list/dictionary if not set?

Also I feel this error message is invalid because I would believe `NoReverseMatch` to be thrown if the link is invalid, not if it was missing args or kwargs.

Reply:
Yeah if I do `dcim:manufacturer` without a `kwargs = {"slug":"arista"}`, it will throw a NoReverseMatch because  it needs to have a slug field specified.
<img width="1491" alt="Screen Shot 2022-08-19 at 1 48 33 PM" src="https://user-images.githubusercontent.com/46973263/185677771-c56e1810-addf-4d76-86cf-deae7174ad39.png">

I do specify a kwargs attribute tho, the link shows up.
<img width="948" alt="Screen Shot 2022-08-19 at 1 49 12 PM" src="https://user-images.githubusercontent.com/46973263/185677880-0eaacfad-5c21-4c79-9e85-2bf7d51d061a.png">
<img width="1010" alt="Screen Shot 2022-08-19 at 1 49 56 PM" src="https://user-images.githubusercontent.com/46973263/185678051-67d03667-d87a-446a-8756-2d6a323186e3.png">

Reply:
But you'll also get a `NoReverseMatch` if the URL pattern is nonexistent or otherwise invalid, right?